docroot/sites/all/modules/features/forms/forms.install

```diff
@@ -166,6 +166,46 @@ function forms_update_7013() {
 }
 
 /**
+ * Remove all traces of entityform type check_your_eligibility (emhrm)
+ */
+function forms_update_7014() {
+  // 1. Remove the submissions - complete; see previous update hook.
+
+  // 2. Delete the field instances
+  $fields_info = field_info_instances('entityform', 'check_your_eligibility');
+  $delete_count = 0;
+  foreach ($fields_info as $field_in_form => $field_array) {
+    $field_name = $field_array['field_name'];
+    if ($instance = field_info_instance('entityform', $field_name,
+      'check_your_eligibility')) {
+        field_delete_instance($instance);
+        $delete_count++;
+    }
+  }
+
+  // 3. Delete fields that aren't used elsewhere
```

Comment:
@carolineHecht remember the DRY principle (Don't Repeat Yourself), and refactor this section into a loop. Also, since field_delete_field also deletes the instances, you could move the new loop you create into step 2, then move step 2 to 3 and it will improve performance.